### PR TITLE
Added mypy

### PR DESCRIPTION
**Pull request by mstefferson**
# Context

[mypy](https://github.com/python/mypy) is an awesome tool for static type checking. Adding to cookiecutter.

# Changes

* Added `mypy`

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/docker/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+mypy==0.790
```
Comment (resdntalien) [inline {{ cookiecutter.repo_name }}/docker/requirements.txt:1]: We should push this upstream into the Orbyter image and keep the requirements here clean.

**Review by resdntalien — CHANGES_REQUESTED** [review]
Love the addition. Can you push the mypy upstream into Orbyter?

**Comment by rameshvs** [comment]
I like this, but unfortunately most of the numeric/ML packages (numpy, pandas, sklearn) have spotty typing support. I'm all in favor of adding it, but we should make sure it's common knowledge among our team and among orbyter users that this is only a partial measure.

File: {{ cookiecutter.repo_name }}/docker/requirements.txt (hunk 1 above)
Comment (mstefferson) [reply {{ cookiecutter.repo_name }}/docker/requirements.txt:1]: You aren't going to let me be lazy are you :-) ? Luckily, I was able to A.D. to add mypy to orbyter:3.2

image

**Review by resdntalien — APPROVED** [review]
LGTM.

**Pull request merged**